packages/@o3r/rules-engine/src/engine/helpers/filter-ruleset-event.operator.spec.ts

```diff
@@ -134,68 +147,66 @@ describe('Filter rulesets event operator', () => {
   }, {});
 
   const firstValue = rulesets.reduce<Record<string, RulesetExecutor>>((accRuleset, ruleset) => {
-    accRuleset[ruleset.id] = new RulesetExecutor(ruleset, {retrieveOrCreateFactStream: () => of(undefined), operators} as any);
+    accRuleset[ruleset.id] = new RulesetExecutor(ruleset, { retrieveOrCreateFactStream: () => of(undefined), operators } as any);
     return accRuleset;
   }, {});
 
   const rulesetsMapSubject$ = new BehaviorSubject<Record<string, RulesetExecutor>>(firstValue);
 
-  test('should consider only first ruleset', (done) => {
-
-    rulesetsMapSubject$.pipe(
-      filterRulesetsEventStream(['ruleset1'])
-    ).subscribe(data => {
-      expect(data.length).toBe(2);
-      done();
+  test('should consider only first ruleset', () => {
+    return new Promise((done) => {
+      rulesetsMapSubject$.pipe(
+        filterRulesetsEventStream(['ruleset1'])
+      ).subscribe((data) => {
+        expect(data.length).toBe(2);
+        done();
+      });
     });
```

Comment:
can't we do something like this:
```suggestion
  test('should consider only first ruleset', async () => {
    const data = await firstValueFrom(rulesetsMapSubject$.pipe(
      filterRulesetsEventStream(['ruleset1'])
    );
    expect(data.length).toBe(2);
```
?